From 8724878308f103cec6e68e9fd1f7af9b4f080ba2 Mon Sep 17 00:00:00 2001
From: Tianshu Qiu <tian.shu.qiu@intel.com>
Date: Fri, 20 Apr 2018 12:20:19 +0800
Subject: [PATCH 1889/2313] media: intel-ipu4: isys: Adapt to new isys css API

Message for Open Source:
New ISYS API adds two bits to specify whether
capture_ack and capture_done are sent from FW.
This patch changes ISYS driver to adapt to this
change.

Message for Internal:
[Issue/Feature] New ISYS API adds two bits to specify whether
capture_ack and capture_done are sent from FW.
[Root Cause/Changes] Changes ISYS driver to adapt to the
FW change.

Change-Id: I8ac3c6486148d57066a1e223ecad3b83696976f4
Signed-off-by: Tianshu Qiu <tian.shu.qiu@intel.com>
Reviewed-by: Coding Style
Approver: CaiyingX Geng <caiyingx.geng@intel.com>
Signed-off-by: Meng Wei <wei.meng@intel.com>
---
 drivers/media/pci/intel/ipu-fw-isys.c                   | 7 +++++++
 drivers/media/pci/intel/ipu-fw-isys.h                   | 4 ++++
 drivers/media/pci/intel/ipu-isys-queue.c                | 8 ++++++++
 drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c   | 2 ++
 drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c | 2 ++
 5 files changed, 23 insertions(+)

diff --git a/drivers/media/pci/intel/ipu-fw-isys.c b/drivers/media/pci/intel/ipu-fw-isys.c
index 9093e99..e52f2af 100644
--- a/drivers/media/pci/intel/ipu-fw-isys.c
+++ b/drivers/media/pci/intel/ipu-fw-isys.c
@@ -217,4 +217,11 @@ void ipu_fw_isys_dump_frame_buff_set(struct device *dev,
 	dev_dbg(dev, "send_irq_eof 0x%x\n", buf->send_irq_eof);
 	dev_dbg(dev, "send_resp_sof 0x%x\n", buf->send_resp_sof);
 	dev_dbg(dev, "send_resp_eof 0x%x\n", buf->send_resp_eof);
+#if defined(CONFIG_VIDEO_INTEL_IPU4) || defined(CONFIG_VIDEO_INTEL_IPU4P)
+	dev_dbg(dev, "send_irq_capture_ack 0x%x\n", buf->send_irq_capture_ack);
+	dev_dbg(dev, "send_irq_capture_done 0x%x\n", buf->send_irq_capture_done);
+#endif
+#ifdef IPU_OTF_SUPPORT
+	dev_dbg(dev, "frame_counter 0x%x\n", buf->frame_counter);
+#endif
 }
diff --git a/drivers/media/pci/intel/ipu-fw-isys.h b/drivers/media/pci/intel/ipu-fw-isys.h
index d5a7e90..2a10721 100644
--- a/drivers/media/pci/intel/ipu-fw-isys.h
+++ b/drivers/media/pci/intel/ipu-fw-isys.h
@@ -678,6 +678,10 @@ struct ipu_fw_isys_frame_buff_set_abi {
 	struct ipu_fw_isys_param_pin_abi process_group_light;
 	u8 send_irq_sof;
 	u8 send_irq_eof;
+#if defined(CONFIG_VIDEO_INTEL_IPU4) || defined(CONFIG_VIDEO_INTEL_IPU4P)
+	u8 send_irq_capture_ack;
+	u8 send_irq_capture_done;
+#endif
 	u8 send_resp_sof;
 	u8 send_resp_eof;
 	u8 reserved;
diff --git a/drivers/media/pci/intel/ipu-isys-queue.c b/drivers/media/pci/intel/ipu-isys-queue.c
index 09ddd0e..8936d5b 100644
--- a/drivers/media/pci/intel/ipu-isys-queue.c
+++ b/drivers/media/pci/intel/ipu-isys-queue.c
@@ -495,6 +495,10 @@ void ipu_isys_buffer_list_to_ipu_fw_isys_frame_buff_set(
 	set->send_resp_sof = 1;
 	set->send_irq_eof = 1;
 	set->send_resp_eof = 1;
+#if defined(CONFIG_VIDEO_INTEL_IPU4) || defined(CONFIG_VIDEO_INTEL_IPU4P)
+	set->send_irq_capture_ack = 1;
+	set->send_irq_capture_done = 1;
+#endif
 
 	list_for_each_entry(ib, &bl->head, head) {
 		if (ib->type == IPU_ISYS_VIDEO_BUFFER) {
@@ -1327,6 +1331,10 @@ int ipu_isys_req_prepare(struct media_device *mdev,
 	set->send_resp_sof = 1;
 	set->send_irq_eof = 1;
 	set->send_resp_eof = 1;
+#if defined(CONFIG_VIDEO_INTEL_IPU4) || defined(CONFIG_VIDEO_INTEL_IPU4P)
+	set->send_irq_capture_ack = 1;
+	set->send_irq_capture_done = 1;
+#endif
 
 	spin_lock_irqsave(&ireq->lock, flags);
 
diff --git a/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c b/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c
index 71161b6..c0e5ec8 100644
--- a/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c
+++ b/drivers/media/pci/intel/ipu4/ipu4-css/libintel-ipu4.c
@@ -346,6 +346,8 @@ static void frame_buff_set_abi_to_api(
 
 	api->send_irq_sof = abi->send_irq_sof;
 	api->send_irq_eof = abi->send_irq_eof;
+	api->send_irq_capture_ack = abi->send_irq_capture_ack;
+	api->send_irq_capture_done = abi->send_irq_capture_done;
 }
 
 int ipu_fw_isys_complex_cmd(struct ipu_isys *isys,
diff --git a/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c b/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c
index c2f8113..8d62b95 100644
--- a/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c
+++ b/drivers/media/pci/intel/ipu4/ipu4p-css/libintel-ipu4p.c
@@ -345,6 +345,8 @@ static void frame_buff_set_abi_to_api(
 
 	api->send_irq_sof = abi->send_irq_sof;
 	api->send_irq_eof = abi->send_irq_eof;
+	api->send_irq_capture_ack = abi->send_irq_capture_ack;
+	api->send_irq_capture_done = abi->send_irq_capture_done;
 }
 
 int ipu_fw_isys_complex_cmd(struct ipu_isys *isys,
-- 
2.7.4

